src/utils.js

```
@@ -542,23 +628,74 @@ export function gcd(a, b) {
+	async init() {}
+	/**
+	 * Creates an instance.
+	 * @template {AsyncFactory} T
+	 * @template {any[]} P
+	 * @this {{ new(...params: P): T }}
+	 * @param {P} params
+	 * @returns {Promise<T>}
+	 */
+	static async new(...params) {
+		let classes = getClassInheritance(this);
+		classes.forEach(c => AsyncFactory.#allowedConstructors.add(c));
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)